admin-dev/themes/default/js/bundle/product/form.js

```diff
@@ -929,25 +929,59 @@ var form = (function() {
         });
       };
 
-      /** On event "tokenfield:createtoken" : stop event if its not a typehead result */
+      /** On event "tokenfield:createtoken" : check values are valid if its not a typehead result */
       $('#form_step3_attributes').on('tokenfield:createtoken', function(e) {
-        if (!e.attrs.data && e.handleObj.origType !== 'tokenfield:createtoken') {
-          return false;
+        if (!e.attrs.data)
+          if (e.handleObj.origType !== 'tokenfield:createtoken') {
+            return false;
+          }
```

Comment:
```suggestion
        if (!e.attrs.data && e.handleObj.origType !== 'tokenfield:createtoken') {
          return false;
        }
```

Reply:
Could you explain why you split the if statement?

Reply:
The statement was split as the next section of the code only needs to run if the data attributes have not been set. The revision was suggested earlier in this converstion. However I have noticed that with the change including the dropping of braces it actually doesn't work as intended. I will restructure it and push a new version which will make the decision clearer.